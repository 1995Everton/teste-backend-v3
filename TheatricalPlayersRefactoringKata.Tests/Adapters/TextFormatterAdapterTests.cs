using ApprovalTests;
using ApprovalTests.Reporters;
using System.Collections.Generic;
using TheatricalPlayersRefactoringKata.Application.Adapters;
using TheatricalPlayersRefactoringKata.Entities;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests.Adapters
{
    public class TextFormatterAdapterTests
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void Format_ShouldReturnFormattedStatement()
        {
            // Configura um cenário onde uma declaração é formatada em texto.
            // Verifica se a declaração formatada está correta.

            // Arrange
            var formatter = new TextFormatterAdapter();
            var statement = new Statement
            {
                Customer = "John Doe",
                Items = new List<StatementItem>
                    {
                        new StatementItem { Name = "Hamlet", AmountOwed = 400.00m, Seats = 55 },
                        new StatementItem { Name = "As You Like It", AmountOwed = 300.00m, Seats = 35 },
                        new StatementItem { Name = "Othello", AmountOwed = 500.00m, Seats = 40 }
                    },
                TotalAmountOwed = 1200.00m,
                TotalEarnedCredits = 30
            };

            // Act
            var result = formatter.Format(statement);

            // Assert
            Approvals.Verify(result);
        }
    }
}
