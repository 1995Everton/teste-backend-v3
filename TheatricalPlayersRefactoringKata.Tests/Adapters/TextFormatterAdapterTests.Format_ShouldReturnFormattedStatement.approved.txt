Statement for John Doe
  Hamlet: $400.00 (55 seats)
  As You Like It: $300.00 (35 seats)
  Othello: $500.00 (40 seats)
Amount owed is $1,200.00
You earned 30 credits
